### PR TITLE
[TRTLLM-13948][feat] Set DeepSeekV3 to use Python KV-cache transceiver V2 by default

### PR DESCRIPTION
**Pull request by asfiyab-nvidia**
## Description

Built on top of https://github.com/NVIDIA/TensorRT-LLM/pull/16344/ and https://github.com/NVIDIA/TensorRT-LLM/pull/16669 (per-model transceiver runtime auto selection): opt DeepSeekV3.2/R1/V3-Lite into the Python KV-cache transceiver by default.

DeepseekV32ForCausalLM and DeepseekV3ForCausalLM now overrides get_preferred_transceiver_runtime() to return "PYTHON". Disaggregated deployments that leave transceiver_runtime at its default auto (backend NIXL/DEFAULT) use the Python transceiver (KvCacheTransceiverV2) out of the box. Explicit user settings always take precedence, and incompatible backends fall back to the C++ transceiver per the https://github.com/NVIDIA/TensorRT-LLM/pull/16164 resolution semantics. No effect on non-disaggregated deployments.

Additionally set the below configs to use the CPP Transceiver while PR https://github.com/NVIDIA/TensorRT-LLM/pull/16834/ addresses the perf regression observed during migration of the transceiver from CPP to PYTHON. 
```
gb300_deepseek-r1-fp4_8k1k_con4096_ctx1_dep4_gen1_dep16_eplb0_mtp1_ccb-NIXL.yaml
gb300_deepseek-r1-fp4_128k8k_con256_ctx1_pp4_gen1_dep8_eplb0_mtp1_ccb-NIXL.yaml
```

## Test Coverage

- Updates the test class `TestGlm5TransceiverPreference` to `TestDeepseekTransceiverPreference`
- Updates the following tests: `test_preference_per_architecture`
- Replaces the following tests: `test_no_config_defers_to_cpp` -> `test_prefers_python_without_config` and `test_glm5_resolves_auto_to_python_on_nixl` -> `test_deepseek_resolves_auto_to_python_on_nixl`

## PR Checklist

Please review the following before submitting your PR:
- PR description clearly explains what and why. If using CodeRabbit's summary, please make sure it makes sense.
- PR Follows [TRT-LLM CODING GUIDELINES](https://github.com/NVIDIA/TensorRT-LLM/blob/main/CODING_GUIDELINES.md) to the best of your knowledge.
- Test cases are provided for new code paths (see [test instructions](https://github.com/NVIDIA/TensorRT-LLM/tree/main/tests#1-how-does-the-ci-work))
- If PR introduces API changes, an appropriate PR label is added - either `api-compatible` or `api-breaking`. For `api-breaking`, include `BREAKING` in the PR title.
- Any new dependencies have been scanned for license and vulnerabilities
- [CODEOWNERS](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/CODEOWNERS) updated if ownership changes
- Documentation updated as needed
- Update [tava architecture diagram](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/tava_architecture_diagram.md) if there is a significant design change in PR.
- The reviewers assigned automatically/manually are appropriate for the PR.

- [x] Please check this after reviewing the above items as appropriate for this PR.

## GitHub Bot Help

To see a list of available CI bot commands, please comment `/bot help`.

## Dev Engineer Review

- DeepSeek V3, V3.2, and GLM MoE DSA now consistently prefer the Python KV-cache transceiver under automatic runtime selection.
- Explicit runtime settings remain respected, while incompatible backends can fall back appropriately; non-disaggregated deployments are unaffected.
- Updated unit tests cover supported architectures, missing configuration, and DeepSeek-on-NIXL resolution.
- Four DeepSeek R1 disaggregated benchmark configurations explicitly select `transceiver_runtime: CPP` for both context and generation workers, preserving the intended performance baseline.
- No apparent API, error-handling, or configuration consistency issues were identified.

## QA Engineer Review

- Modified test code: transceiver preference tests in `tests/unittest/llmapi/test_llm_args.py`; the suite was renamed/reworked from GLM-5-focused coverage to DeepSeek/GLM coverage.
- Updated coverage includes:
  - Python runtime preference for supported DeepSeek V3, DeepSeek V3.2, and GLM architectures.
  - Python preference when no pretrained configuration is provided.
  - Automatic DeepSeek-on-NIXL runtime resolution.
- No `tests/integration/test_lists/`, `test-db/`, or `qa/` files were modified, so CI/manual test-list coverage is not represented.
- Reported affected unit tests passed; remaining CI failures were unrelated to this PR.

**Verdict: needs follow-up** — test-list coverage data is unavailable.

### CODE REVIEW

**Comment by asfiyab-nvidia** [comment]
/bot run --extra-stage "A10-PyTorch-1,A10-PyTorch-2,A10-PyTorch-3"

File: tests/unittest/llmapi/test_llm_args.py [hunk 1]
```diff
@@ -3404,43 +3399,33 @@ def _pretrained_config(architectures, model_type):
...
+    def test_deepseek_resolves_auto_to_python_on_nixl(self):
+        """DeepseekV3ForCausalLM on NIXL adopts the Python transceiver from 'auto'."""
         from tensorrt_llm._torch.models.modeling_deepseekv3 import \
             DeepseekV3ForCausalLM
         args = TorchLlmArgs(
             model="/tmp/dummy_model",
             cache_transceiver_config=CacheTransceiverConfig(backend="NIXL"),
         )
-        cfg = self._pretrained_config(["GlmMoeDsaForCausalLM"], "glm_moe_dsa")
+        cfg = self._pretrained_config(["DeepseekV3ForCausalLM"], "deepseek_v3")
```
[the remark this answers is not in the document]
Comment (asfiyab-nvidia) [reply tests/unittest/llmapi/test_llm_args.py]: Addressed

File: tensorrt_llm/_torch/models/modeling_deepseekv3.py [hunk 2]
```diff
@@ -1897,25 +1897,14 @@ class DeepseekV3ForCausalLM(SpecDecOneEngineForCausalLM[DeepseekV3Model,
...
-        if pretrained_config is None:
-            return None
-        architectures = getattr(pretrained_config, 'architectures', None) or []
-        # model_type is checked as a fallback: it is 'glm_moe_dsa' on GLM
-        # checkpoints until __init__ rewrites it to 'deepseek_v32'.
-        if ("GlmMoeDsaForCausalLM" in architectures or getattr(
-                pretrained_config, 'model_type', None) == 'glm_moe_dsa'):
-            return "PYTHON"
-        return None
+        return "PYTHON"
```
Comment (nv-xtf) [inline tensorrt_llm/_torch/models/modeling_deepseekv3.py:1907]: This change silently flips two GB300 perf configs that were deliberately reverted to the V1 transceiver. When [#16669](https://github.com/NVIDIA/TensorRT-LLM/pull/16669) migrated the DeepSeek R1/V3.2 disagg perf cases to transceiver_runtime: PYTHON, gb300_deepseek-r1-fp4_128k8k_con256_ctx1_pp4... and gb300_deepseek-r1-fp4_8k1k_con4096_ctx1_dep4... (4 files across tests/scripts/perf-sanity/disaggregated/ and tests/scripts/perf/disaggregated/) were intentionally left unpinned, relying on the then-current "auto → C++" resolution to stay on V1. This PR changes what auto resolves to for DeepSeek, so those 4 configs will silently switch to the V2 path they were reverted away from. If the GB300 issue is not yet resolved, please pin transceiver_runtime: CPP in those 4 files as part of this PR; if it is resolved, please state that in the description.

**Review by nv-xtf — CHANGES_REQUESTED** [review]

File: tensorrt_llm/_torch/models/modeling_deepseekv3.py (hunk 2 above)
Comment (asfiyab-nvidia) [reply tensorrt_llm/_torch/models/modeling_deepseekv3.py:1907]: Thanks for the catch. Pushed the change and updated the description accordingly

**Review by nv-xtf — APPROVED** [review]
LGTM

**Comment by Shixiaowei02** [comment]
Could you please confirm whether the DeepSeek unit tests have been fully replaced? Thanks!

**Comment by asfiyab-nvidia** [comment]
Tests affected in this PR have passed - https://prod.blsm.nvidia.com/sw-tensorrt-top-1/job/LLM/job/main/job/L0_MergeRequest_PR/50424/testReport/A10-PyTorch-2.unittest.llmapi.test_llm_args/TestDeepseekTransceiverPreference/

Remaining tests are unrelated to the changes made

**Comment by asfiyab-nvidia** [comment]
> Could you please confirm whether the DeepSeek unit tests have been fully replaced? Thanks!

Here is a link to the replaced tests: https://prod.blsm.nvidia.com/sw-tensorrt-top-1/job/LLM/job/main/job/L0_MergeRequest_PR/50424/testReport/A10-PyTorch-2.unittest.llmapi.test_llm_args/TestDeepseekTransceiverPreference/

**Review by 2ez4bz — APPROVED** [review]

**Review by lori-ren — APPROVED** [review]

File: tests/scripts/perf-sanity/disaggregated/gb300_deepseek-r1-fp4_128k8k_con256_ctx1_pp4_gen1_dep8_eplb0_mtp1_ccb-NIXL.yaml [hunk 3]
```diff
@@ -64,6 +64,7 @@ worker_config:
     cache_transceiver_config:
       max_tokens_in_buffer: 131104
       backend: NIXL
+      transceiver_runtime: CPP
```
Comment (dc3671) [inline tests/scripts/perf-sanity/disaggregated/gb300_deepseek-r1-fp4_128k8k_con256_ctx1_pp4_gen1_dep8_eplb0_mtp1_ccb-NIXL.yaml:67]: @hyukn @chenfeiz0326 Do you think we should rename or use other way to separate these four configs? Otherwise these may be mixed with all other configs and we never know it's different in `transceiver_runtime`
